scala/lambdas/preingest-tdr-aggregator/src/main/scala/uk/gov/nationalarchives/preingesttdraggregator/Lambda.scala

```diff
@@ -28,13 +29,12 @@ class Lambda extends RequestHandler[SQSEvent, Unit]:
   given DASFNClient[IO] = DASFNClient[IO]()
   given DADynamoDBClient[IO] = DADynamoDBClient[IO]()
 
-  override def handleRequest(input: SQSEvent, context: Context): Unit =
-    ConfigSource.default
-      .loadF[IO, Config]()
-      .flatMap { config =>
-        Aggregator[IO].aggregate(config, groupCacheAtomicCell, input.getRecords.asScala.toList, context.getRemainingTimeInMillis)
-      }
-      .unsafeRunSync()
+  override def handleRequest(input: SQSEvent, context: Context): SQSBatchResponse = {
+    for {
+      config <- ConfigSource.default.loadF[IO, Config]()
+      results <- Aggregator[IO].aggregate(config, groupCacheAtomicCell, input.getRecords.asScala.toList, context.getRemainingTimeInMillis)
```

Comment:
Maybe this could be "potentialFailures" or "BatchItemFailures" or something?

Reply:
`potentialFailures` is good because there might be none.